File: java-parser/FunctionValidate.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.Arrays;
import java.util.HashMap;
import org.antlr.v4.gui.TreeViewer;
import org.antlr.v4.runtime.ANTLRInputStream;
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CodePointCharStream;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.tree.ParseTree;

public class FunctionValidate {
    public static void main(String[] args) throws IOException
    {
        boolean print_tree = false;
        int file_index = 0;
        if (args.length == 0)
        {
            System.out.println("Antlr Java parser.");
            return;
        }
        else
        {
            for (int i = 0; i < args.length; ++i)
            {
                if (args[i] == "-t")
                {
                    print_tree = true;
                }
                else
                {
                    file_index = i;
                }
            }
        }
        int exceptions = 0;
        ErrorListener errorListener = new ErrorListener();
        ParseTree tree = null;
        CommonTokenStream tokens = null;
        try
        {
            File file = new File(args[file_index]);
            FileInputStream input = null;
            input = new FileInputStream(file);
            ANTLRInputStream str = new ANTLRInputStream(input);
            JavaLexer lexer = new JavaLexer(str);
            tokens = new CommonTokenStream(lexer);
            JavaParser parser = new JavaParser(tokens);
            parser.removeErrorListeners();
            parser.addErrorListener(errorListener);
            tree = parser.compilationUnit();
        }
        catch (Exception e)
        {
            exceptions++;
        }
        System.out.println("errors " + errorListener.num_errors + " exceptions " + exceptions);
        Output output = new Output();
        if (tree != null) System.out.println(output.OutputTree(tree, tokens));
    }
}

File: csharp-parser/FunctionValidate.cs
namespace Test
{
    using System;
    using System.IO;
    using Antlr4.Runtime;
    using Antlr4.Runtime.Tree;

    public class FunctionValidate {
        public static void Main(String[] args)
        {
            bool print_tree = false;
            int file_index = 0;
            if (args.Length == 0)
            {
                System.Console.WriteLine("Antlr Java parser.");
                return;
            }
            else
            {
                for (int i = 0; i < args.Length; ++i)
                {
                    if (args[i] == "-t")
                    {
                        print_tree = true;
                    }
                    else
                    {
                        file_index = i;
                    }
                }
            }
            int exceptions = 0;
	        var errorListener = new ErrorListener<IToken>();
	        IParseTree tree = null;
            CommonTokenStream tokens = null;
            try
            {
                var input = File.OpenText(args[file_index]);
                var str = new AntlrInputStream(input);
                JavaLexer lexer = new JavaLexer(str);
                tokens = new CommonTokenStream(lexer);
                var parser = new JavaParser(tokens);
                parser.RemoveErrorListeners();
                parser.AddErrorListener(errorListener);
                tree = parser.compilationUnit();
            }
            catch (Exception e)
            {
                exceptions++;
            }
            System.Console.WriteLine("errors " + errorListener.num_errors + " exceptions " + exceptions);
	        if (tree != null) System.Console.WriteLine(tree.OutputTree(tokens));
        }
    }
}
